networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -263,11 +263,18 @@ def allocate_hosting_port(self, context, router_id, port_db, network_type,
         # If this is a router interface, the VLAN comes from APIC.
         # If it's the gateway, the VLAN comes from the segment ID
         if port_db.get('device_owner') == DEVICE_OWNER_ROUTER_GW:
-            return super(AciVLANTrunkingPlugDriver,
-                         self).allocate_hosting_port(
-                             context, router_id,
-                             port_db, network_type, hosting_device_id
-                         )
+            ext_dict, net = self._get_external_network_dict(context, port_db)
```

Comment:
We want the following behavior:
1) if it's not an opflex network type, call the superclass to get the segmentation ID
2) if it's an opflex network, get the segmentation ID from the config. If it's not present, we should throw an exception.
The current code should be changed to address number 2 (i.e. no exception is being thrown the way that it's currently written).